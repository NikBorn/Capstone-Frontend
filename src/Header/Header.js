import React, { Component } from 'react';

class Header extends Component {

  render() {
    return (
      <header className='header'>
        <h1 className='header-title'>Concert Tracker</h1>
        <nav className='topNav'>
          <ul>
            <li>Find Bands </li> 
            <li>Find Venues </li> 
            <li>Find locations </li> 
          </ul>
        </nav> 
        <button className='sign-in sign-up'>Sign Up or Sign In</button>
      </header>
    );
  }

}

export default Header;


