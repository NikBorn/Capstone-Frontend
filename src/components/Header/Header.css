
.header {
  background-color: #303F68;
  color: rgb(244, 244, 248);
  border-bottom: 4px solid rgb(244, 244, 248);
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 100px;
}

.header-title a {
  color: rgb(244, 244, 248);
}

.header-content {
  flex-grow: 2;
  display: flex;
  align-items: center;
  justify-content: space-between
}

@media screen and (max-width: 1030px) {

  .header {
    background-color: magenta;
    padding: 0 40px;
  }
}

@media screen and (max-width: 910px) {

  .header {
    background-color: lime;
    /* flex-direction: column; */
  }

  ul {
    margin: 0;
  }

  li {
    margin: 0 0 6px 0;;
  }

  .SearchBar-input,
  .searh-button {
    margin: 10px 0;
  }
}

@media screen and (max-width: 910px) {
  .header-content {
    flex-direction: column;
  }

  li {
    margin: 6px 0
  }
}

