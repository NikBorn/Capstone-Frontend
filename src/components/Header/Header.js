import React, { Component } from 'react';
import SignInSignUp from '../SignInSignUp/SignInSignUp';
import SearchBar from '../SearchBar/SearchBar';
import NavBar from '../NavBar/NavBar';

class Header extends Component {

  render() {
    return (
      <header className='header'>
        <h1 className='header-title'>Concert Tracker</h1>
        <NavBar />
        <SearchBar />
        <SignInSignUp />
      </header>
    );
  }

}

export default Header;


