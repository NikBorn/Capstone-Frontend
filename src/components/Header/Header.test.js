import { shallow, configure } from 'enzyme';
import Header from './Header';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe('Header snapshot', () => {

  it('should always match the snapshot', () => {

    const wrapper = shallow(<Header />);

    expect(wrapper).toMatchSnapshot();
  });
});