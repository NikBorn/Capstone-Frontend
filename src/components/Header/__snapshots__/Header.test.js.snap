// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="header-title"
>
          <Link
                    replace={false}
                    to="/"
          >
                    Concert Tracker
          </Link>
</h1>,
        <div
          className="header-content"
>
          <Connect(NavBar) />
          <withRouter(Connect(SearchBar)) />
          <withRouter(Connect(SignInSignUp)) />
</div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/"
>
            Concert Tracker
</Link>,
          "className": "header-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Concert Tracker",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Concert Tracker",
          "type": [Function],
        },
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(NavBar) />,
            <withRouter(Connect(SearchBar)) />,
            <withRouter(Connect(SignInSignUp)) />,
          ],
          "className": "header-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="header-title"
>
            <Link
                        replace={false}
                        to="/"
            >
                        Concert Tracker
            </Link>
</h1>,
          <div
            className="header-content"
>
            <Connect(NavBar) />
            <withRouter(Connect(SearchBar)) />
            <withRouter(Connect(SignInSignUp)) />
</div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
>
              Concert Tracker
</Link>,
            "className": "header-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Concert Tracker",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Concert Tracker",
            "type": [Function],
          },
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(NavBar) />,
              <withRouter(Connect(SearchBar)) />,
              <withRouter(Connect(SignInSignUp)) />,
            ],
            "className": "header-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
